src/Framework/Sdk/SdkLogger.cs

```diff
@@ -15,5 +17,32 @@ public abstract class SdkLogger
         /// <param name="message">Message string.</param>
         /// <param name="messageImportance">Optional message importances. Default to low.</param>
         public abstract void LogMessage(string message, MessageImportance messageImportance = MessageImportance.Low);
+
+        /// <summary>
+        /// Logs that an event.
+        /// </summary>
+        /// <param name="args">An array of arguments to log with the event.</param>
+        public virtual void LogEvent(params object[] args)
+        {
+            MSBuildEventSource.Log.SdkResolverEvent(args);
+        }
+
+        /// <summary>
+        /// Logs that an event when an operation has started.
+        /// </summary>
+        /// <param name="args">An array of arguments to log with the event.</param>
+        public virtual void LogEventStart(params object[] args)
+        {
+            MSBuildEventSource.Log.SdkResolverEventStart(args);
+        }
+
+        /// <summary>
+        /// Logs that an event when an operation has completed.
+        /// </summary>
+        /// <param name="args">An array of arguments to log with the event.</param>
+        public virtual void LogEventStop(params object[] args)
+        {
+            MSBuildEventSource.Log.SdkResolverEventStop(args);
+        }
```

Comment:
Thank you for making the change. After re-reviewing I'm wondering if maybe these methods shouldn't be in another class altogether.
- They also "log" but it's a different kind of logging than what `LogMessage` is supposed to do. It could be confusing for the caller.
- Do you expect a component to override these? What justifies the `virtual` keyword?

I'm probably overthinking it, the API is public but a not a commonly used one. Feel free to ignore me.

Reply:
> After re-reviewing I'm wondering if maybe these methods shouldn't be in another class altogether.

I sort of agree, but I also think there's no better place.  I could add a new class and pass it to resolvers but that would be a breaking change.  Do you propose a better spot to have these?

> Do you expect a component to override these? What justifies the virtual keyword?

No but if anyone is implementing this class they'll be broken if the new methods are not `virtual`.  As far as I know, the only implementations are unit tests.  When this was originally created, we decided to use classes instead of interfaces so we could add stuff like this without it being a breaking change.  Of course now you can add default implementations of interfaces...

Reply:
> No but if anyone is implementing this class they'll be broken if the new methods are not `virtual`

First I was like "that's not the case, adding non-virtual methods doesn't break implementors" but it totally makes sense on semantic level if the goal is to enable mocking. All clear now, I think it's fine as is. Thank you!